modules/cucumber-reportportal-formatter.js

```diff
@@ -592,24 +628,19 @@ const createRPFormatterClass = (config) => {
         status: isFailed ? 'failed' : 'passed',
         endTime: reportportal.helpers.now(),
       });
-      context.scenarioStatus = 'failed';
       context.scenarioId = null;
 
       const featureUri = event.sourceLocation.uri;
-      if (!event.result.retried) {
-        context.scenariosCount[featureUri].done++;
-      }
-      const { total, done } = context.scenariosCount[featureUri];
-      if (done === total) {
-        const featureStatus = context.failedScenarios[featureUri] > 0 ? 'failed' : 'passed';
-        reportportal.finishTestItem(pickleDocuments[featureUri].featureId, {
-          status: featureStatus,
-          endTime: reportportal.helpers.now(),
-        });
-      }
+      featureData[featureUri].featureStatus = context.failedScenarios[featureUri] > 0 ? 'failed' : 'passed';
     }
 
     onTestRunFinished() {
+      Object.keys(featureData).forEach(feature => {
```

Comment:
Why you are finishing all features at the end of the test run? When executed sequentially, their actual completion time will differ.

Reply:
Could we make some time to discuss this on the ReportPortal slack or something? I had very good reasons for all the changes I made and many of them were because there was core Cucumber JS functionality which wasn't being taken into account and therefore causing issues with report portal reporting.

Reply:
Hi!
Ok, let's discuss this.
You can write me personally on RP slack (Ilya Hancharyk).